subsys/fs/fat_fs.c

```diff
@@ -458,6 +458,31 @@ static int fatfs_unmount(struct fs_mount_t *mountp)
 	return translate_error(res);
 }
 
+#if defined(CONFIG_FILE_SYSTEM_MKFS)
+
+static int fatfs_mkfs(struct fs_mount_t *mountp)
+{
+#if defined(CONFIG_FS_FATFS_MOUNT_MKFS)
```

Comment:
This is not correct, the Kconfig used here is to enable/disable mkfs during mount, it should not affect mkfs, which is independent from mount.

Reply:
I have removed that.